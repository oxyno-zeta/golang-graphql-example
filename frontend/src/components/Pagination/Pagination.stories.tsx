import React from 'react';
import { StoryFn, Meta } from '@storybook/react';
import { withRouter } from 'storybook-addon-react-router-v6';
import Pagination, { Props } from './Pagination';

export default {
  title: 'Components/Pagination',
  component: Pagination,
  decorators: [withRouter],
  args: {
    pageInfo: {
      hasNextPage: true,
      endCursor: 'fake-end',
    },
    maxPaginationSize: 10,
  },
} as Meta<typeof Pagination>;

export const Playground: StoryFn<typeof Pagination> = function C(args: Props) {
  return <Pagination {...args} handleFirstPage={undefined} handleNextPage={undefined} handlePreviousPage={undefined} />;
};
Playground.parameters = {
  reactRouter: {
    routePath: '/route',
  },
};

export const OnlyNextPage: StoryFn<typeof Pagination> = function C(args: Props) {
  return (
    <Pagination
      {...args}
      handleFirstPage={undefined}
      handleNextPage={undefined}
      handlePreviousPage={undefined}
      pageInfo={{ hasNextPage: true, hasPreviousPage: false, endCursor: 'fake-end' }}
    />
  );
};
OnlyNextPage.parameters = {
  reactRouter: {
    routePath: '/route',
  },
};

export const OnlyPreviousPage: StoryFn<typeof Pagination> = function C(args: Props) {
  return (
    <Pagination
      {...args}
      handleFirstPage={undefined}
      handleNextPage={undefined}
      handlePreviousPage={undefined}
      pageInfo={{ hasNextPage: false, hasPreviousPage: true, startCursor: 'fake-start' }}
    />
  );
};
OnlyPreviousPage.parameters = {
  reactRouter: {
    routePath: '/route',
  },
};

export const AllEnabled: StoryFn<typeof Pagination> = function C(args: Props) {
  return (
    <Pagination
      {...args}
      handleFirstPage={undefined}
      handleNextPage={undefined}
      handlePreviousPage={undefined}
      pageInfo={{ hasNextPage: true, hasPreviousPage: true, startCursor: 'fake-start', endCursor: 'fake-end' }}
    />
  );
};
AllEnabled.parameters = {
  reactRouter: {
    routePath: '/route',
  },
};

export const AllEnabledWithFunctions: StoryFn<typeof Pagination> = function C(args: Props) {
  return (
    <Pagination
      {...args}
      pageInfo={{ hasNextPage: true, hasPreviousPage: true, startCursor: 'fake-start', endCursor: 'fake-end' }}
    />
  );
};
AllEnabledWithFunctions.parameters = {
  reactRouter: {
    routePath: '/route',
  },
};
