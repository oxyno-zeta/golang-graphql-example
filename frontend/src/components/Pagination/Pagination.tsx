import React, { ForwardRefExoticComponent, RefAttributes } from 'react';
import { Link, useHref, useLocation, createSearchParams, useSearchParams, LinkProps } from 'react-router-dom';
import ChevronRightIcon from '@mui/icons-material/ChevronRight';
import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
import KeyboardDoubleArrowLeftIcon from '@mui/icons-material/KeyboardDoubleArrowLeft';
import Tooltip from '@mui/material/Tooltip';
import IconButton, { IconButtonProps } from '@mui/material/IconButton';
import Toolbar, { ToolbarProps } from '@mui/material/Toolbar';
import { useTranslation } from 'react-i18next';
import { PageInfoModel } from '../../models/general';
import { getAllSearchParams } from '../../utils/urlSearchParams';
import { cleanPaginationSearchParams } from '../../utils/pagination';

interface Props {
  pageInfo: PageInfoModel;
  maxPaginationSize: number;
  handleFirstPage?: () => void | undefined;
  handlePreviousPage?: () => void | undefined;
  handleNextPage?: () => void | undefined;
  toolbarProps?: ToolbarProps;
  firstIconButtonProps?: IconButtonProps;
  previousIconButtonProps?: IconButtonProps;
  nextIconButtonProps?: IconButtonProps;
}

const defaultProps = {
  handleFirstPage: undefined,
  handlePreviousPage: undefined,
  handleNextPage: undefined,
  toolbarProps: {},
  firstIconButtonProps: {},
  previousIconButtonProps: {},
  nextIconButtonProps: {},
};

type IconButtonInternalProps = {
  to?: string;
  component?: ForwardRefExoticComponent<LinkProps & RefAttributes<HTMLAnchorElement>>;
  onClick?: () => void;
};

function Pagination({
  maxPaginationSize,
  pageInfo,
  handleFirstPage,
  handlePreviousPage,
  handleNextPage,
  toolbarProps,
  firstIconButtonProps,
  nextIconButtonProps,
  previousIconButtonProps,
}: Props) {
  // Setup translate
  const { t } = useTranslation();
  // Get search params
  const [searchParams] = useSearchParams();
  // Setup location
  const location = useLocation();

  // Build first page props
  let firstPageProps: IconButtonInternalProps = {
    to: useHref({
      pathname: location.pathname,
      search: createSearchParams(getAllSearchParams(cleanPaginationSearchParams(searchParams))).toString(),
    }),
    component: Link,
  };
  // Check if handle first page is declared
  if (handleFirstPage) {
    firstPageProps = {
      onClick: handleFirstPage,
    };
  }

  // Build previous page props
  let previousPageProps: IconButtonInternalProps = {
    to: useHref({
      pathname: location.pathname,
      search: createSearchParams({
        ...getAllSearchParams(cleanPaginationSearchParams(searchParams)),
        before: pageInfo.startCursor || '',
        last: maxPaginationSize.toString(),
      }).toString(),
    }),
    component: Link,
  };
  // Check if handle previous page is declared
  if (handlePreviousPage) {
    previousPageProps = {
      onClick: handlePreviousPage,
    };
  }

  // Build next page props
  let nextPageProps: IconButtonInternalProps = {
    to: useHref({
      pathname: location.pathname,
      search: createSearchParams({
        ...getAllSearchParams(cleanPaginationSearchParams(searchParams)),
        after: pageInfo.endCursor || '',
        first: maxPaginationSize.toString(),
      }).toString(),
    }),
    component: Link,
  };
  // Check if handle next page is declared
  if (handleNextPage) {
    nextPageProps = {
      onClick: handleNextPage,
    };
  }

  return (
    <Toolbar variant="dense" {...toolbarProps}>
      <div style={{ marginLeft: 'auto' }}>
        <Tooltip title={<>{t('common.firstPageAction')}</>}>
          <span>
            <IconButton disabled={!pageInfo.hasPreviousPage} {...firstPageProps} {...firstIconButtonProps}>
              <KeyboardDoubleArrowLeftIcon />
            </IconButton>
          </span>
        </Tooltip>
        <Tooltip title={<>{t('common.previousPageAction')}</>}>
          <span>
            <IconButton disabled={!pageInfo.hasPreviousPage} {...previousPageProps} {...nextIconButtonProps}>
              <ChevronLeftIcon />
            </IconButton>
          </span>
        </Tooltip>
        <Tooltip title={<>{t('common.nextPageAction')}</>}>
          <span>
            <IconButton disabled={!pageInfo.hasNextPage} {...nextPageProps} {...previousIconButtonProps}>
              <ChevronRightIcon />
            </IconButton>
          </span>
        </Tooltip>
      </div>
    </Toolbar>
  );
}

Pagination.defaultProps = defaultProps;

export default Pagination;
