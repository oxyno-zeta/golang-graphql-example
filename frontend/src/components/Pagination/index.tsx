import Pagination, { Props } from './Pagination';

export default Pagination;
export type { Props };
