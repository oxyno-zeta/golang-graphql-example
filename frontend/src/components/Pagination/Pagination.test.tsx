import React from 'react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { fireEvent, render } from '@testing-library/react';
// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';

import Pagination, { Props as PaginationProps } from './Pagination';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

interface Props {
  readonly paginationProps: PaginationProps;
}

function TestComponent({ paginationProps }: Props) {
  return (
    <MemoryRouter initialEntries={['/route']}>
      <Routes>
        <Route element={<Pagination {...paginationProps} />} path="/route" />
      </Routes>
    </MemoryRouter>
  );
}

describe('Pagination', () => {
  it('should display 3 disable links when no next or previous page are present', async () => {
    const { container } = render(
      <TestComponent
        paginationProps={{
          maxPaginationSize: 10,
          pageInfo: { hasNextPage: false, hasPreviousPage: false },
        }}
      />,
    );

    const allA = container.querySelectorAll('a');
    expect(allA).toHaveLength(3);
    // Loop over
    allA.forEach((item) => {
      expect(item).toHaveClass('Mui-disabled');
    });
    expect(container).toMatchSnapshot();
  });

  it('should display only next page link as enabled with correct url', async () => {
    const { container } = render(
      <TestComponent
        paginationProps={{
          maxPaginationSize: 10,
          pageInfo: { hasNextPage: true, endCursor: 'fake-end', hasPreviousPage: false },
        }}
      />,
    );

    const allA = container.querySelectorAll('a');
    expect(allA).toHaveLength(3);
    expect(allA[0]).toHaveClass('Mui-disabled');
    expect(allA[1]).toHaveClass('Mui-disabled');
    expect(allA[2]).not.toHaveClass('Mui-disabled');
    expect(allA[2]).toHaveAttribute('href', '/route?after=fake-end&first=10');
    expect(container).toMatchSnapshot();
  });

  it('should display only previous page links as enabled with correct url', async () => {
    const { container } = render(
      <TestComponent
        paginationProps={{
          maxPaginationSize: 10,
          pageInfo: { hasNextPage: false, startCursor: 'fake-start', hasPreviousPage: true },
        }}
      />,
    );

    const allA = container.querySelectorAll('a');
    expect(allA).toHaveLength(3);
    expect(allA[0]).not.toHaveClass('Mui-disabled');
    expect(allA[0]).toHaveAttribute('href', '/route');
    expect(allA[1]).not.toHaveClass('Mui-disabled');
    expect(allA[1]).toHaveAttribute('href', '/route?before=fake-start&last=10');
    expect(allA[2]).toHaveClass('Mui-disabled');
    expect(container).toMatchSnapshot();
  });

  it('should display previous and next page links as enabled with correct url', async () => {
    const { container } = render(
      <TestComponent
        paginationProps={{
          maxPaginationSize: 10,
          pageInfo: {
            hasNextPage: true,
            startCursor: 'fake-start',
            hasPreviousPage: true,
            endCursor: 'fake-end',
          },
        }}
      />,
    );

    const allA = container.querySelectorAll('a');
    expect(allA).toHaveLength(3);
    expect(allA[0]).not.toHaveClass('Mui-disabled');
    expect(allA[0]).toHaveAttribute('href', '/route');
    expect(allA[1]).not.toHaveClass('Mui-disabled');
    expect(allA[1]).toHaveAttribute('href', '/route?before=fake-start&last=10');
    expect(allA[2]).not.toHaveClass('Mui-disabled');
    expect(allA[2]).toHaveAttribute('href', '/route?after=fake-end&first=10');
    expect(container).toMatchSnapshot();
  });

  it('should display only next page button as enabled and clickable', async () => {
    let clicked = false;
    const { container } = render(
      <TestComponent
        paginationProps={{
          maxPaginationSize: 10,
          pageInfo: { hasNextPage: true, endCursor: 'fake-end', hasPreviousPage: false },
          onNextPage: () => {
            clicked = true;
          },
        }}
      />,
    );

    const allA = container.querySelectorAll('a');
    expect(allA).toHaveLength(2);
    expect(allA[0]).toHaveClass('Mui-disabled');
    expect(allA[1]).toHaveClass('Mui-disabled');
    const allButtons = container.querySelectorAll('button');
    expect(allButtons).toHaveLength(1);
    expect(allButtons[0]).not.toHaveClass('Mui-disabled');
    expect(allButtons[0]).not.toHaveAttribute('href', '/route?after=fake-end&first=10');
    expect(container).toMatchSnapshot();

    expect(fireEvent.click(allButtons[0])).toBeTruthy();
    expect(clicked).toBeTruthy();
  });

  it('should display only previous page buttons as enabled and clickable', async () => {
    let previousPageClicked = false;
    let firstPageClicked = false;
    const { container } = render(
      <TestComponent
        paginationProps={{
          maxPaginationSize: 10,
          pageInfo: { hasNextPage: false, startCursor: 'fake-start', hasPreviousPage: true },
          onPreviousPage: () => {
            previousPageClicked = true;
          },
          onFirstPage: () => {
            firstPageClicked = true;
          },
        }}
      />,
    );

    const allA = container.querySelectorAll('a');
    expect(allA).toHaveLength(1);
    expect(allA[0]).toHaveClass('Mui-disabled');
    const allButtons = container.querySelectorAll('button');
    expect(allButtons).toHaveLength(2);
    expect(allButtons[0]).not.toHaveClass('Mui-disabled');
    expect(allButtons[0]).not.toHaveAttribute('href', '/route');
    expect(allButtons[1]).not.toHaveClass('Mui-disabled');
    expect(allButtons[1]).not.toHaveAttribute('href', '/route?before=fake-start&last=10');
    expect(container).toMatchSnapshot();

    expect(fireEvent.click(allButtons[0])).toBeTruthy();
    expect(firstPageClicked).toBeTruthy();
    expect(fireEvent.click(allButtons[1])).toBeTruthy();
    expect(previousPageClicked).toBeTruthy();
  });

  it('should display next and previous page buttons as enabled and clickable', async () => {
    let nextPageClicked = false;
    let previousPageClicked = false;
    let firstPageClicked = false;
    const { container } = render(
      <TestComponent
        paginationProps={{
          maxPaginationSize: 10,
          pageInfo: {
            hasNextPage: true,
            endCursor: 'fake-end',
            startCursor: 'fake-start',
            hasPreviousPage: true,
          },
          onNextPage: () => {
            nextPageClicked = true;
          },
          onPreviousPage: () => {
            previousPageClicked = true;
          },
          onFirstPage: () => {
            firstPageClicked = true;
          },
        }}
      />,
    );

    const allA = container.querySelectorAll('a');
    expect(allA).toHaveLength(0);
    const allButtons = container.querySelectorAll('button');
    expect(allButtons).toHaveLength(3);
    expect(allButtons[0]).not.toHaveClass('Mui-disabled');
    expect(allButtons[0]).not.toHaveAttribute('href', '/route');
    expect(allButtons[1]).not.toHaveClass('Mui-disabled');
    expect(allButtons[1]).not.toHaveAttribute('href', '/route?before=fake-start&last=10');
    expect(allButtons[2]).not.toHaveClass('Mui-disabled');
    expect(allButtons[2]).not.toHaveAttribute('href', '/route?after=fake-end&first=10');
    expect(container).toMatchSnapshot();

    expect(fireEvent.click(allButtons[0])).toBeTruthy();
    expect(firstPageClicked).toBeTruthy();
    expect(fireEvent.click(allButtons[1])).toBeTruthy();
    expect(previousPageClicked).toBeTruthy();
    expect(fireEvent.click(allButtons[2])).toBeTruthy();
    expect(nextPageClicked).toBeTruthy();
  });
});
