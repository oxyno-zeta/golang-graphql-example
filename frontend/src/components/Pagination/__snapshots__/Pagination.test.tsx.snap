// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should display 3 disable links when no next or previous page are present 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-10ltm04-MuiToolbar-root"
  >
    <div
      style="margin-left: auto;"
    >
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <a
          aria-disabled="true"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          href="/route"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.41,7.41L17,6L11,12L17,18L18.41,16.59L13.83,12L18.41,7.41M12.41,7.41L11,6L5,12L11,18L12.41,16.59L7.83,12L12.41,7.41Z"
            />
          </svg>
        </a>
      </span>
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <a
          aria-disabled="true"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          href="/route?before=&last=10"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
            />
          </svg>
        </a>
      </span>
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <a
          aria-disabled="true"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          href="/route?after=&first=10"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
            />
          </svg>
        </a>
      </span>
    </div>
  </div>
</div>
`;

exports[`Pagination should display next and previous page buttons as enabled and clickable 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-10ltm04-MuiToolbar-root"
  >
    <div
      style="margin-left: auto;"
    >
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.41,7.41L17,6L11,12L17,18L18.41,16.59L13.83,12L18.41,7.41M12.41,7.41L11,6L5,12L11,18L12.41,16.59L7.83,12L12.41,7.41Z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </span>
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </span>
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </span>
    </div>
  </div>
</div>
`;

exports[`Pagination should display only next page button as enabled and clickable 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-10ltm04-MuiToolbar-root"
  >
    <div
      style="margin-left: auto;"
    >
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <a
          aria-disabled="true"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          href="/route"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.41,7.41L17,6L11,12L17,18L18.41,16.59L13.83,12L18.41,7.41M12.41,7.41L11,6L5,12L11,18L12.41,16.59L7.83,12L12.41,7.41Z"
            />
          </svg>
        </a>
      </span>
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <a
          aria-disabled="true"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          href="/route?before=&last=10"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
            />
          </svg>
        </a>
      </span>
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </span>
    </div>
  </div>
</div>
`;

exports[`Pagination should display only next page link as enabled with correct url 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-10ltm04-MuiToolbar-root"
  >
    <div
      style="margin-left: auto;"
    >
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <a
          aria-disabled="true"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          href="/route"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.41,7.41L17,6L11,12L17,18L18.41,16.59L13.83,12L18.41,7.41M12.41,7.41L11,6L5,12L11,18L12.41,16.59L7.83,12L12.41,7.41Z"
            />
          </svg>
        </a>
      </span>
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <a
          aria-disabled="true"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          href="/route?before=&last=10"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
            />
          </svg>
        </a>
      </span>
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <a
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          href="/route?after=fake-end&first=10"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </span>
    </div>
  </div>
</div>
`;

exports[`Pagination should display only previous page buttons as enabled and clickable 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-10ltm04-MuiToolbar-root"
  >
    <div
      style="margin-left: auto;"
    >
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.41,7.41L17,6L11,12L17,18L18.41,16.59L13.83,12L18.41,7.41M12.41,7.41L11,6L5,12L11,18L12.41,16.59L7.83,12L12.41,7.41Z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </span>
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </span>
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <a
          aria-disabled="true"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          href="/route?after=&first=10"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
            />
          </svg>
        </a>
      </span>
    </div>
  </div>
</div>
`;

exports[`Pagination should display only previous page links as enabled with correct url 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-10ltm04-MuiToolbar-root"
  >
    <div
      style="margin-left: auto;"
    >
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <a
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          href="/route"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.41,7.41L17,6L11,12L17,18L18.41,16.59L13.83,12L18.41,7.41M12.41,7.41L11,6L5,12L11,18L12.41,16.59L7.83,12L12.41,7.41Z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </span>
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <a
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          href="/route?before=fake-start&last=10"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </span>
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <a
          aria-disabled="true"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          href="/route?after=&first=10"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
            />
          </svg>
        </a>
      </span>
    </div>
  </div>
</div>
`;

exports[`Pagination should display previous and next page links as enabled with correct url 1`] = `
<div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-10ltm04-MuiToolbar-root"
  >
    <div
      style="margin-left: auto;"
    >
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <a
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          href="/route"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18.41,7.41L17,6L11,12L17,18L18.41,16.59L13.83,12L18.41,7.41M12.41,7.41L11,6L5,12L11,18L12.41,16.59L7.83,12L12.41,7.41Z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </span>
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <a
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          href="/route?before=fake-start&last=10"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </span>
      <span
        class=""
        data-mui-internal-clone-element="true"
      >
        <a
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          href="/route?after=fake-end&first=10"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.59,16.58L13.17,12L8.59,7.41L10,6L16,12L10,18L8.59,16.58Z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </span>
    </div>
  </div>
</div>
`;
